common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -980,6 +981,40 @@ def _should_apply_cached_metadata(self, item, depth):
         return apply_cached_metadata
 
     @autoretry_read()
+    def get_course_summaries(self, **kwargs):
+        """
+        Returns a list of `CourseSummary`. This accepts an optional parameter of 'org' which
+        will apply an efficient filter to only get courses with the specified ORG
+        """
+        def extract_course_summary(course):
+            """
+            Extract course information from the course block for mongo.
+            """
+            return {
+                field: course['metadata'][field]
+                for field in CourseSummary.course_info_fields
+                if field in course['metadata']
+            }
+
+        course_org_filter = kwargs.get('org')
+        meta_only = {'metadata': True}
+
+        if course_org_filter:
+            course_records = self.collection.find({'_id.category': 'course', '_id.org': course_org_filter}, meta_only)
+        else:
+            course_records = self.collection.find({'_id.category': 'course'}, meta_only)
```

Comment:
Nit: I'd prefer a single PyMongo call here (`self.collection.find`), with conditional logic building the submitted query above.